src/DynamicRupture/Output/Base.cpp

```diff
@@ -214,77 +216,329 @@ void Base::updateElementwiseOutput() {
+      local.waveSpeedsMinus = &((local.layer->var(drDescr->waveSpeedsMinus))[local.ltsId]);
+
+      auto faultInfo = faultInfos[faceIndex];
+
+      real dofsPlus[tensor::Q::size()]{};
+      getDofs(dofsPlus, faultInfo.element, faultInfo.side);
+
+      real dofsMinus[tensor::Q::size()]{};
+      getDofs(dofsMinus, faultInfo.neighborElement, faultInfo.neighborSide);
+
+      auto* initStresses = local.layer->var(drDescr->initialStressInFaultCS);
+      auto* initStress = initStresses[local.ltsId][local.nearestGpIndex];
+
+      local.mu = (local.layer->var(drDescr->mu))[local.ltsId][local.nearestGpIndex];
+      local.sXY = initStress[3];
+      local.sXZ = initStress[5];
+      local.p0 = initStress[0];
+      local.pf = this->computePf();
+
+      const auto* const normal = outputData.faultDirections[i].faceNormal;
+      const auto* const tangent1 = outputData.faultDirections[i].tangent1;
+      [[maybe_unused]] const auto* const tangent2 = outputData.faultDirections[i].tangent2;
+      auto* const strike = outputData.faultDirections[i].strike;
+      [[maybe_unused]] const auto* dip = outputData.faultDirections[i].dip;
+
+      auto* phiPlusSide = outputData.basisFunctions[i].plusSide.data();
+      auto* phiMinusSide = outputData.basisFunctions[i].minusSide.data();
+
+      seissol::dynamicRupture::kernel::computeFaceAlignedValues kernel;
+      kernel.Tinv = outputData.glbToFaceAlignedData[i].data();
+
+      kernel.Q = dofsPlus;
+      kernel.basisFunctionsAtPoint = phiPlusSide;
+      kernel.QAtPoint = local.faceAlignedValuesPlus;
+      kernel.execute();
+
+      kernel.Q = dofsMinus;
+      kernel.basisFunctionsAtPoint = phiMinusSide;
+      kernel.QAtPoint = local.faceAlignedValuesMinus;
+      kernel.execute();
+
+      this->computeLocalStresses();
+      real strength = this->computeLocalStrength();
+      this->computeLocalTraction(strength);
+
+      seissol::dynamicRupture::kernel::rotateInitStress alignAlongDipAndStrikeKernel;
+      alignAlongDipAndStrikeKernel.stressRotationMatrix =
+          outputData.stressGlbToDipStrikeAligned[i].data();
+      alignAlongDipAndStrikeKernel.reducedFaceAlignedMatrix =
+          outputData.stressFaceAlignedToGlb[i].data();
+
+      std::array<real, 6> tmpVector{};
+      tmpVector[0] = local.p;
+      tmpVector[1] = local.yyStress;
+      tmpVector[2] = local.zzStress;
+      tmpVector[3] = local.xyTraction;
+      tmpVector[4] = local.yzStress;
+      tmpVector[5] = local.xzTraction;
+
+      alignAlongDipAndStrikeKernel.initialStress = tmpVector.data();
+      std::array<real, 6> rotatedTraction{};
+      alignAlongDipAndStrikeKernel.rotatedStress = rotatedTraction.data();
+      alignAlongDipAndStrikeKernel.execute();
+
+      tmpVector[0] = local.p;
+      tmpVector[1] = local.yyStress;
+      tmpVector[2] = local.zzStress;
+      tmpVector[3] = local.xyStress;
+      tmpVector[4] = local.yzStress;
+      tmpVector[5] = local.xzStress;
```

Comment:
This is the same as above. `initialStress` is not changed by the kernel, so we can leave this out?

Reply:
I lost the context of this note. Could you address it in a new PR

Reply:
I can't find it in the new PR :/